artifacts-wg/charter/charter.md

```diff
@@ -0,0 +1,49 @@
+# Purpose
+In the distributed world of cloud-native computing different artifacts and packages are used to transport configuration and code for the many services and capabilities that comprise and support workloads and applications. As an example, today ArtifactHub advertises that it can crawl about 20 such artifact types [1]. These "cloud-native" bundles add to the previous proliferation of package and artifact types for software dependencies such as Maven and npm and for system packages like deb and rpm.
+
+This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully learned and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development by users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
+
+To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects offering packaging, distribution and deployment mechanisms for bundles of configuration and code.
+
+**The WG will:**  
+* **Develop common patterns**  
+*  **Advocate for innovative projects**  
+*  **Gather end user feedback**
+**for packaging, discovery, distribution, and deployment mechanisms for** **cloud-native software artifacts.**
+
+# Alignment with TAG App Delivery and CNCF
+TAG App Delivery's charter [2] calls on the TAG to develop best practices, foster collaboration between projects, improve interoperability and propose new initiatives and projects related to the following domains which this working group would target.
+
+* Application bundling and deployment
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on official specifications in the Open Containers Initiative (OCI) itself.
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/artifacts
+* Develop a pattern for dynamically discovering existing Artifact schemas
+This in contrast to asking all schema developers to publish media types via IANA.
+* Contribute to a common data model for Artifacts in OCI
```

Comment:
There's going to be more than one data model for artifact content. This implies they will all have a single model.

Reply:
If this is getting into defining a new OCI manifest type, OCI is already planning to look at a replacement for the artifact manifest after the image-spec 1.1 release. It's probably better to do that over in OCI.

Reply:
I think that's the reason that this working group was originally pitched to the OCI maintainers. It would be unfortunate if the next OCI data model was not compatible with the search API. With consideration for that prior attempt at establishing this working group under OCI, do you have any suggestions for combining these efforts or at least ensuring this working group's output is a consideration in the development of the next OCI data model?

Reply:
Our focus right now is on the 1.1 release. Once that's done, merging efforts would be great.

Reply:
This could be rephrased a bit to focus on the need to package various types of artifacts vs defining a single model (depending on the level where we implement this, we may still have a single model, we may leverage an existing model, or there may be multiple models).

```suggestion
* Define data models for storing Artifacts in OCI
```

Reply:
I would caution activities that define a deliverable outcome as that's what your success will be hinged upon.
Defining OCI for artefacts is going to be a long process of getting people to agree with each other. 
Perhaps you can save yourself with wording here
```suggestion
Identify an exemplar data model for storing artefacts in OCI
```

Reply:
@sudo-bmitch @AlexsJones, based on your suggestions and the feedback from @justincormack, I've tried to better address this in the revised goals.